DbService/src/DbValTool.cc

```diff
@@ -0,0 +1,242 @@
+
+#include "Offline/DbService/inc/DbValTool.hh"
+#include "cetlib_except/exception.h"
+
+namespace mu2e {
+
+  //**************************************************
+
+int DbValTool::findPid(std::string purpose) const {
+
+  if (purpose.empty()) return -1;
+
+  for (auto const& pp : _valcache.valPurposes().rows()) {
+    if (pp.name() == purpose) return pp.pid();
+  }
+  return -1;
+}
```

Comment:
Same question as for DbSet.cc  It is safe to count on the caller to test return values for validity?  Is there well defined and safe behaviour if they do not?

Again, if this is for internal use only we can decide to keep it as is.

Reply:
The code should not throw, but I can change to "bool findByName(int& tid)" if you prefer.  I don't imagine non-expert users, but you never know for sure.

Reply:
Please do this.